### ISSUE TITLE
Use random cryptographic secrets for client application

### ISSUE THREAD

**Issue by sebastienros** [OrchardCMS/Orchard2#786]
- Users don't need to generate custom ones
- More secure
- Prevents user lookup

@jersiovic @PinpointTownes

**Comment by jersiovic** [comment]
+1

**Comment by PinpointTownes** [comment]
I'm all for it, but don't count on me to submit a PR, I'm currently overbooked :sweat_smile:

**Comment by sebastienros** [comment]
@PinpointTownes I didn't count on it at all. Just for awareness that it won't be forgotten. We are already thankful you take the time to look at the code.

**Comment by jersiovic** [comment]
+1 to to what @sebastienros has pointed
@PinpointTownes I will do it ... when I find time

**Comment by sebastienros** [comment]
Isn't it done already? I think I remember someone who was complaining they could not set it because it was generated.

**Comment by PinpointTownes** [comment]
> Isn't it done already?

Nope:

https://github.com/OrchardCMS/OrchardCore/blob/98d0ab89f5a29f56efa62c2d1dfdaab095204abd/src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AdminController.cs#L270-L278

https://github.com/OrchardCMS/OrchardCore/blob/98d0ab89f5a29f56efa62c2d1dfdaab095204abd/src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AdminController.cs#L324-L363

**Comment by sebastienros** [comment]
Ok, I'll try to do it and have you guys review it. thanks.

**Comment by PinpointTownes** [comment]
FYI, OpenIddict RC1 will be released this week. Let me know if you're interested in updating the OpenID module to use the latest bits. There are many changes, but I can assist you on Skype if necessary.

**Comment by sebastienros** [comment]
Not this week, as we are preparing for nuget release but right after that for sure. And I assume it means releasing on Nuget?

**Comment by PinpointTownes** [comment]
Yep. If you push your packages before I do so with mine, please make sure you don't accidentally push the OpenIddict packages.

**Comment by sebastienros** [comment]
Haven't you reserved the names? Also I only push the artifacts that are in /src, and I don't see why we'd have a .nupkg from you there.

**Comment by PinpointTownes** [comment]
> Haven't you reserved the names?

Nope, is it easy to do?

**Comment by sebastienros** [comment]
Create a script that pushes a fake package which is unlisted using the nuget command. That's how I did for all our packages.

**Comment by PinpointTownes** [comment]
I'm going to remove the "password validation" stuff as part of my next PR. We'll still be able to introduce a secret generation feature in the future.

**Comment by gvkries** [comment]
@kevinchalet Isn't this fixed by #16467 now?

**Comment by kevinchalet** [comment]
@gvkries yep! One 8-year-old ticket less it is, then 😄
